src/owner.rs

```diff
@@ -1,4 +1,32 @@
-//! Owner pattern
+//! Owner pattern provides methods to query, manage and transfer ownership
+//! of the contract.
+//!
+//! An owned contract is initialized with an owner. The account that
+//! currently owns the contract is the "current owner". It can propose
+//! an account to transfer ownership to. This proposed account is the
+//! "proposed owner". If the proposed owner accepts the transfer, it becomes
+//! the current owner. The current owner can also renounce ownership of the
+//! contract.
+//!
+//! Note: There is no way to recover ownership of a renounced contract.
+//!
+//! The pattern consists of methods in [`Owner`] and [`OwnerExternal`]. The
+//! latter exposes methods externally and can be called by other contracts.
+//! The provided [derive_macro][`near_contract_tools_macros::Owner`]
+//! derives default implementation both these traits.
+//!
+//! # Safety
+//! The default implementation throws an error or shows unexpected behaviour,
+//! if the following invariants are not met.
+//!
+//! * The owner root storage key is not used or modified. The default key is "~o".
```

Comment:
The default key is only the default if the corresponding macro is used. There isn't currently a default with just the raw trait, though that's something worth considering, since right now it is non-obvious where the default key is declared.

Also, I think it should say something like "The owner root storage _slot_ is not..."

Same goes for the other components.

Reply:
Now using "storage slot" terminology.

The docs makes it clear that the default implementation is done by a macro. It is implied that it is the default implementation done by the macro that decides the value of the key. Is there something else that should be mentioned here?